Configuration/scripts/makeEfficiencyPlots.py

```diff
@@ -262,11 +266,24 @@ def MakeOneHist(dirName, histogramName):
         if arguments.normalizeToUnitArea:
             yAxisLabel = yAxisLabel + " (Unit Area Norm.)"
 
-        #Histogram = ROOT.TGraphAsymmErrors(NumHistogramObj,DenHistogramObj)
-        if arguments.noTGraph or Histogram.Class().InheritsFrom("TH2"):
+        #check if bin content is consistent
+        for i in range(1,Histogram.GetNbinsX()+1):
+            if Histogram.GetBinContent(i) > DenHistogram.GetBinContent(i):
+                DenHistogram.SetBinContent(i,Histogram.GetBinContent(i))
+
+        #HistogramClone and HistogramClones only used for ratio plot
+        HistogramClone = Histogram.Clone()
+        HistogramClone.SetDirectory(0)
+        HistogramClone.Divide(DenHistogram)
+        Nbins = HistogramClone.GetNbinsX()
+        MaxXValue = HistogramClone.GetXaxis().GetBinLowEdge(Nbins+1)
+        MinXValue = HistogramClone.GetXaxis().GetBinLowEdge(1)
+
+        #this Histogram becomes the main TEfficiency
+        if Histogram.Class().InheritsFrom("TH2"):
             Histogram.Divide(DenHistogram)
         else:
-            Histogram = TGraphAsymmErrors(Histogram,DenHistogram)
+            Histogram = TEfficiency(Histogram,DenHistogram)
```

Comment:
I am not recommending a code change here, but in case someone has specialized needs we should remember that these divide functions use default methods:

https://root.cern.ch/doc/master/classTEfficiency.html (c.f. section IV)

Perhaps just a link to that in comment is enough.

Reply:
added this comment as you suggest. as i wrote in the comment, the default method should usually be the right one, at least as far as the uncertainties are concerned.